plugins/comparative-adapters/src/BlastTabularAdapter/BlastTabularAdapter.ts

```diff
@@ -0,0 +1,256 @@
+import { readConfObject } from '@jbrowse/core/configuration'
+import {
+  BaseFeatureDataAdapter,
+  BaseOptions,
+} from '@jbrowse/core/data_adapters/BaseAdapter'
+import { getSubAdapterType } from '@jbrowse/core/data_adapters/dataAdapterCache'
+import PluginManager from '@jbrowse/core/PluginManager'
+import { ObservableCreate } from '@jbrowse/core/util/rxjs'
+import { doesIntersect2, Feature, isGzip, Region } from '@jbrowse/core/util'
+import { openLocation } from '@jbrowse/core/util/io'
+
+import { parseLineByLine } from '../util'
+import BlastTabularAdapterConfigType from './configSchema'
+import SyntenyFeature from '../SyntenyFeature'
+
+type BlastTabularAdapterConfig = ReturnType<
+  typeof BlastTabularAdapterConfigType.create
+>
+
+// Blast output column names/descriptions taken from
+// https://www.ncbi.nlm.nih.gov/books/NBK279684/#_appendices_Options_for_the_commandline_a_
+interface BlastRecord {
+  /** Query Seq-id */
+  qseqid: string
+  /** Subject Seq-id */
+  sseqid: string
+  /** Percentage of identical matches */
+  pident: number
+  /** Alignment length */
+  length: number
+  /** Number of mismatches */
+  mismatch: number
+  /** Number of gap openings */
+  gapopen: number
+  /** Start of alignment in query */
+  qstart: number
+  /** End of alignment in query */
+  qend: number
+  /** Start of alignment in subject */
+  sstart: number
+  /** End of alignment in subject */
+  send: number
+  /** Expect value */
+  evalue: number
+  /** Bit score */
+  bitscore: number
+}
+
+export function parseBlastLine(line: string): BlastRecord | undefined {
+  const [
+    qseqid,
+    sseqid,
+    pident,
+    length,
+    mismatch,
+    gapopen,
+    qstart,
+    qend,
+    sstart,
+    send,
+    evalue,
+    bitscore,
+  ] = line.split('\t')
+
+  if (
+    !(
+      qseqid &&
+      sseqid &&
+      pident &&
+      length &&
+      mismatch &&
+      gapopen &&
+      qstart &&
+      qend &&
+      sstart &&
+      send &&
+      evalue &&
+      bitscore
+    )
+  ) {
+    console.warn('Invalid BLAST line')
+    console.warn(line)
+    return
+  }
+
+  return {
+    qseqid,
+    sseqid,
+    pident: Number.parseFloat(pident),
+    length: Number.parseInt(length, 10),
+    mismatch: Number.parseInt(mismatch, 10),
+    gapopen: Number.parseInt(gapopen, 10),
+    qstart: Number.parseInt(qstart, 10),
+    qend: Number.parseInt(qend, 10),
+    sstart: Number.parseInt(sstart, 10),
+    send: Number.parseInt(send, 10),
+    evalue: Number.parseFloat(evalue),
+    bitscore: Number.parseFloat(bitscore),
+  }
+}
+
+export default class BlastTabularAdapter extends BaseFeatureDataAdapter {
+  private data: Promise<BlastRecord[]>
+
+  public static capabilities = ['getFeatures', 'getRefNames']
+
+  constructor(
+    public config: BlastTabularAdapterConfig,
+    public getSubAdapter?: getSubAdapterType,
+    public pluginManager?: PluginManager,
+  ) {
+    super(config, getSubAdapter, pluginManager)
+    this.data = this.setup(config)
```

Comment:
I would avoid calling async operations in the constructor like this, it is difficult to provide error handling this way. you can possibly reference other ways that this has been done with "setupPre" for example, and then any function that needs to download the dataset calls another function "setup" that caches the setupPre promise

Reply:
Fixed